CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+cmake_minimum_required(VERSION 3.13)
+
+project(spectatord)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+add_compile_options(-fno-omit-frame-pointer -fsanitize=address)
```

Comment:
We should scope the `-fsanitize=address` to only the debug build so we can run tests and catch issues then but our release build should avoid it since it has a significant overhead.

Maybe something like: 
```
add_compile_options(-fno-omit-frame-pointer)
add_compile_options(
    "$<$<CONFIG:DEBUG>:-fsanitize=address>"
)
```

Reply:
Good catch, will fix.